modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,361 @@
+package org.tessellation.infrastructure.trust.storage
+
+import cats.syntax.option._
+import cats.syntax.partialOrder._
+import cats.syntax.semigroup._
+
+import org.tessellation.cli.method.{delayToUpdateCurrent, snapshotInterval}
+import org.tessellation.domain.trust.storage.{OrdinalTrustMap, PublicTrustMap, TrustMap}
+import org.tessellation.infrastructure.trust.generators.{genPeerLabel, genPeerObservationAdjustmentUpdate, genPeerTrustInfo}
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.schema.generators.peerIdGen
+import org.tessellation.schema.trust._
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.numeric.NonNegative
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustStorageSuite extends SimpleIOSuite with Checkers {
+  test("initial trust storage is empty") {
+    for {
+      store <- TrustStorage.make
+      trust <- store.getTrust
+    } yield expect.all(TrustMap.empty == trust)
+  }
+
+  test("internal trust storage is updated with trust labels") {
+    val gen = Gen.listOfN(4, genPeerObservationAdjustmentUpdate).map(labels => PeerObservationAdjustmentUpdateBatch(labels))
+
+    forall(gen) { labels =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updateTrust(labels)
+
+        expected = labels.updates.map(update => update.id -> TrustInfo().copy(trustLabel = update.trust.value.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          TrustMap.empty == firstTrust,
+          expected == actual
+        )
+    }
+  }
+
+  test("internal trust storage is updated with predicted trusts") {
+    val gen = Gen.mapOfN(4, genPeerLabel)
+
+    forall(gen) { trusts =>
+      for {
+        store <- TrustStorage.make
+        firstTrust <- store.getTrust
+
+        _ <- store.updatePredictedTrust(trusts)
+
+        expected = trusts.view.mapValues(trustValue => TrustInfo().copy(predictedTrust = trustValue.some)).toMap
+        actual <- store.getTrust.map(_.trust)
+      } yield
+        expect.all(
+          TrustMap.empty == firstTrust,
+          expected == actual
+        )
+    }
+  }
+
+  test("internal trust storage is updated with public trust") {
+    val gen = for {
+      peerId <- peerIdGen
+      labels <- Gen.mapOfN(4, genPeerLabel)
+    } yield (peerId, PublicTrust(labels))
+
+    forall(gen) {
+      case (peerId, publicTrust) =>
+        for {
+          store <- TrustStorage.make
+          firstTrust <- store.getTrust
+
+          _ <- store.updatePeerPublicTrustInfo(peerId, publicTrust)
+
+          expected = PublicTrustMap(Map(peerId -> PublicTrust(publicTrust.labels)))
+          actual <- store.getTrust.map(_.peerLabels)
+        } yield
+          expect.all(
+            TrustMap.empty == firstTrust,
+            expected == actual
```

Comment:
aa

Reply:
All equality in the tests I created have been updated to use `===`.